app/src/test/java/org/simple/clinic/newentry/DateOfBirthFormatValidatorTest.kt

```diff
@@ -23,18 +25,46 @@ class DateOfBirthFormatValidatorTest {
     "24/04, INVALID_PATTERN",
     " , INVALID_PATTERN"
   ])
-  fun validate(date: String, expectedResult: Result) {
+  fun `validate (v1)`(date: String, expectedResult: Result) {
     val format = DateTimeFormatter.ofPattern("dd/MM/yyyy", Locale.ENGLISH)
 
     assertThat(DateOfBirthFormatValidator(format).validate(date)).isEqualTo(expectedResult)
   }
 
   @Test
-  fun `validate future date`() {
+  @Parameters(method = "params for v2 validation")
+  fun `validate (v2)`(date: String, expectedResult: DateOfBirthFormatValidator.Result2) {
+    val format = DateTimeFormatter.ofPattern("dd/MM/yyyy", Locale.ENGLISH)
+
+    assertThat(DateOfBirthFormatValidator(format).validate2(date)).isEqualTo(expectedResult)
+  }
+
+  @Suppress("unused")
+  fun `params for v2 validation`(): List<Any> {
+    return listOf(
+        listOf("24/04/1971", Valid(LocalDate.of(1971, 4, 24))),
+        listOf("24-04-1971", Invalid.InvalidPattern),
+        listOf("1971-04-24", Invalid.InvalidPattern),
+        listOf("1971-24-04", Invalid.InvalidPattern),
+        listOf("24/04", Invalid.InvalidPattern),
+        listOf(" ", Invalid.InvalidPattern))
+  }
+
+  @Test
+  fun `validate future date (v1)`() {
     val format = DateTimeFormatter.ofPattern("dd/MM/yyyy", Locale.ENGLISH)
     val dobValidator = DateOfBirthFormatValidator(format)
     val futureDateResult = dobValidator.validate("01/01/3000", nowDate = LocalDate.parse("2018-07-16"))
 
     assertThat(futureDateResult).isEqualTo(Result.DATE_IS_IN_FUTURE)
   }
+
+  @Test
+  fun `validate future date (v2)`() {
+    val format = DateTimeFormatter.ofPattern("dd/MM/yyyy", Locale.ENGLISH)
+    val dobValidator = DateOfBirthFormatValidator(format)
+    val futureDateResult = dobValidator.validate2("01/01/3000", nowDate = LocalDate.parse("2018-07-16"))
```

Comment:
Can you make this a parameterized test and provide a couple more combinations of dates to verify this behaviour?

Reply:
Can I do this separately? Asking because this change wasn't a part of this feature. I simply duplicated the logic for using sealed classes.

Reply:
Sure.